lib/package/sqs/compileMethodsToSqs.test.js

```diff
@@ -573,13 +583,106 @@ describe('#compileMethodsToSqs()', () => {
                 }
               ]
             },
+            PassthroughBehavior: 'NEVER',
             RequestParameters: {
-              'integration.request.querystring.Action': "'SendMessage'",
-              'integration.request.querystring.MessageBody': 'method.request.body',
+              'integration.request.header.Content-Type': "'application/x-www-form-urlencoded'",
               key1: 'value1',
               key2: 'value2'
             },
-            RequestTemplates: { 'application/json': '{statusCode:200}' },
+            RequestTemplates: {
+              'application/json': 'Action=SendMessage&MessageBody=$util.urlEncode($input.body)'
+            },
+            IntegrationResponses: [
+              {
+                StatusCode: 200,
+                SelectionPattern: 200,
+                ResponseParameters: {},
+                ResponseTemplates: {}
+              },
+              {
+                StatusCode: 400,
+                SelectionPattern: 400,
+                ResponseParameters: {},
+                ResponseTemplates: {}
+              },
+              {
+                StatusCode: 500,
+                SelectionPattern: 500,
+                ResponseParameters: {},
+                ResponseTemplates: {}
+              }
+            ]
+          },
+          MethodResponses: [
+            { ResponseParameters: {}, ResponseModels: {}, StatusCode: 200 },
+            { ResponseParameters: {}, ResponseModels: {}, StatusCode: 400 },
+            { ResponseParameters: {}, ResponseModels: {}, StatusCode: 500 }
+          ]
+        }
+      }
+    })
+  })
+
+  it('should return a custom response template for application/json when one is given', () => {
```

Comment:
Should this say "request template" instead of "response template"?

```suggestion
  it('should return a custom request template for application/json when one is given', () => {
```

Reply:
Yes, it should. Will fix.